src/XMakeBuildEngine/Evaluation/Expander.cs

```diff
@@ -3341,12 +3341,12 @@ private static Type GetTypeFromAssembly(string typeName, string candidateAssembl
 
                 // Try to load the assembly with the computed name
 #if FEATURE_GAC
-#pragma warning disable 618
+#pragma warning disable 618, 612
```

Comment:
FYI Marek recently fixed LoadWithPartialName's Obsolete attribute to have a message in mono master.